File: C-Sharp/frmOverride.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	internal partial class frmOverride : System.Windows.Forms.Form
	{
		bool slocalOR;
		int slocalMgrID;

		private void cmdExit_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			this.Close();
		}

		private void frmOverride_Load(System.Object eventSender, System.EventArgs eventArgs)
		{
			txtPassword.Text = "";
		}

		public bool sOverride(ref int sMgrID)
		{
			slocalMgrID = 0;
			slocalOR = false;
			ShowDialog();
			sMgrID = slocalMgrID;
			return slocalOR;
		}

		private void txtPassword_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			ADODB.Recordset rs = default(ADODB.Recordset);
			slocalOR = false;
			 // ERROR: Not supported in C#: OnErrorStatement

			switch (KeyAscii) {
				case 13:
					//Dim user As user
					KeyAscii = 0;
					rs = modRecordSet.getRS(ref "SELECT * FROM Person WHERE (Person_QuickAccess = '" + Strings.Replace(this.txtPassword.Text, "'", "''") + "')");
					if (rs.BOF | rs.EOF) {
						lblError.Text = "Invalid Security Code!";
						txtPassword.SelectionStart = 0;
						txtPassword.SelectionLength = 999;
						goto EventExitSub;
					} else {
						//MsgBox "Login 1"
						if (Convert.ToInt32(rs.Fields("Person_SecurityBit").Value + "0")) {
							if (8192 & rs.Fields("Person_SecurityBit").Value) {
								slocalOR = true;
								slocalMgrID = rs.Fields("PersonID").Value;
								this.Close();
							} else {
								lblError.Text = "Call Your Supervisor!";
								txtPassword.SelectionStart = 0;
								txtPassword.SelectionLength = 999;
								goto EventExitSub;
							}
						}
					}
					break;
				case 27:
					KeyAscii = 0;
					this.Close();
					break;
			}
			EventExitSub:
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}
	}
}

File: VB/4PosBackOffice.NET/frmOverride.vb
Option Strict Off
Option Explicit On
Friend Class frmOverride
	Inherits System.Windows.Forms.Form
	Dim slocalOR As Boolean
	Dim slocalMgrID As Integer
	
	Private Sub cmdExit_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdExit.Click
		Me.Close()
	End Sub
	
	Private Sub frmOverride_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		txtPassword.Text = ""
	End Sub
	
	Public Function sOverride(ByRef sMgrID As Integer) As Boolean
		slocalMgrID = 0
		slocalOR = False
		ShowDialog()
		sMgrID = slocalMgrID
		sOverride = slocalOR
	End Function
	
	Private Sub txtPassword_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtPassword.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		Dim rs As ADODB.Recordset
		slocalOR = False
		On Error Resume Next
		Select Case KeyAscii
			Case 13
				'Dim user As user
				KeyAscii = 0
				rs = getRS("SELECT * FROM Person WHERE (Person_QuickAccess = '" & Replace(Me.txtPassword.Text, "'", "''") & "')")
				If rs.BOF Or rs.EOF Then
					lblError.Text = "Invalid Security Code!"
					txtPassword.SelectionStart = 0
					txtPassword.SelectionLength = 999
					GoTo EventExitSub
				Else
					'MsgBox "Login 1"
					If CInt(rs.Fields("Person_SecurityBit").Value & "0") Then
						If 8192 And rs.Fields("Person_SecurityBit").Value Then
							slocalOR = True
							slocalMgrID = rs.Fields("PersonID").Value
							Me.Close()
						Else
							lblError.Text = "Call Your Supervisor!"
							txtPassword.SelectionStart = 0
							txtPassword.SelectionLength = 999
							GoTo EventExitSub
						End If
					End If
				End If
			Case 27
				KeyAscii = 0
				Me.Close()
		End Select
EventExitSub: 
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
End Class